src/main/java/org/commonwl/view/cwl/CWLTool.java

```diff
@@ -79,12 +79,15 @@ public String getVersion() {
       BufferedReader br = new BufferedReader(isr);
 
       String line;
+      String cwlToolVersion;
       if ((line = br.readLine()) != null) {
         cwlToolVersion = line.substring(line.indexOf(' ') + 1);
-        return cwlToolVersion;
       } else {
-        return "<error getting cwl version>";
+        cwlToolVersion = "<error getting cwl version>";
       }
+      process.destroyForcibly();
```

Comment:
Maybe change it so `process` is initially declared outside the `try/catch`. Then in a finally block do the `if (process != null) ... .destroy...`. So that if there are any IOExceptions or any other errors before we reach this line, this code is called anyway in the `finally` block.